### ISSUE TITLE
Cannot remove_entity() entities with AnimationSystem

### ISSUE THREAD

**Issue by jupart** [kivy/kivent#235]
To reproduce, run example/9_twinkling_stars and call self.gameworld.remove_entity() on one of the created stars. Example:

Change line 117 to `self.ent = `instead of `ent = `.
Add this function:
`def on_touch_down(self, *args):
    self.gameworld.remove_entity(self.ent)`

Traceback:
   File "main.py", line 132, in on_touch_down
     self.gameworld.remove_entity(self.ent)
   File "kivent_core/gameworld.pyx", line 488, in kivent_core.gameworld.GameWorld.remove_entity (kivent_core/gameworld.c:9733)
   File "kivent_core/systems/renderers.pyx", line 611, in kivent_core.systems.renderers.Renderer.remove_component (kivent_core/systems/renderers.c:13096)
   File "kivent_core/managers/resource_managers.pyx", line 127, in kivent_core.managers.resource_managers.ModelManager.unregister_entity_with_model (kivent_core/managers/resource_managers.c:6391)
 KeyError: 'star2-4'

A similar thing happens when loading textures via an atlas, the main atlas .png name (whatever-0.png) gets registered via ModelManager.register_entity_with_model(), but then the current frame in the animation is used as an argument to ModelManager.unregister_entity_with_model(), which isn't in the ._model_register dict.

**Comment by jupart** [comment]
Kovak or anyone else: if you can think of a solution, I don't mind implementing and submitting a PR. Right now, I'm not sure what a good solution would be:

* you don't want to register all images contained in an .atlas with an entity, because an atlas could contain any number of images unrelated to the entity, which would hamper garbage collection (which is what all this is for right?)

* you could leave it how it is and register the atlas name, but we'd need to then store the atlas name in the RenderStruct/render component or something, like how the (<VertexModel>pointer.model)._name is currently stored.

**Comment by SleepProgger** [comment]
IMHO the correct approach would be to unregister the old model and register the new one in the AnimationsSystem update function.

Something like:
```
model_manager.unregister_entity_with_model(
                    render_comp.entity_id,
                    (<VertexModel>render_comp.model)._name)
                render_comp.model = frame_data.model
                model_manager.register_entity_with_model(
                    render_comp.entity_id, self.system_id,
                    (<VertexModel>render_comp.model)._name)
```

**Comment by jupart** [comment]
Your patch works for me!

**Comment by Kovak** [comment]
This is the correct change to make, would anyone like to make a PR for it?

**Comment by SleepProgger** [comment]
I am busy today but could send a PR tomorrow evening.

**Comment by SleepProgger** [comment]
Should be fixed by #236  .?
